charts/common/templates/_images.tpl

```diff
@@ -63,10 +63,10 @@ imagePullSecrets:
 Checks for the imagePullPolicy
 */}}
 {{- define "common.imagePullPolicy" -}}
-  {{- if .Values.global.imagePullPolicy -}}
-    {{- .Values.global.imagePullPolicy -}}
+  {{- if .Values.imagePullPolicy -}}
+    {{- .Values.imagePullPolicy -}}
   {{- else -}}
-    {{- default .Values.imagePullPolicy -}}
+    {{- default .Values.global.imagePullPolicy -}}
```

Comment:
i don't think `default` can be used like this.
`default` should be used like `{{ default default_value some_value }}` and if `some_value` is null, then it uses the default value. you can also use it with a pipe (`|`) but you always need 2 parameters for it.


reference: https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function

Reply:
in this case i don't think the default does anything, and it's also no needed since we always take either the if or the else value anyways

Reply:
thanks, for clearing it out